src/builtins.c

```diff
@@ -1002,10 +998,8 @@ static inline size_t get_checked_fieldindex(const char *name, jl_datatype_t *st,
         jl_value_t *t = jl_type_union(ts, 2);
         jl_type_error("getfield", t, arg);
     }
-    if (mutabl && jl_field_isconst(st, idx)) {
-        jl_errorf("%s: const field .%s of type %s cannot be changed", name,
-                jl_symbol_name((jl_sym_t*)jl_svecref(jl_field_names(st), idx)), jl_symbol_name(st->name->name));
-    }
+    if (mutabl && (!st->name->mutabl || jl_field_isconst(st, idx)))
+        jl_immutable_field_error(st, (jl_sym_t*)jl_svecref(jl_field_names(st), idx));
```

Comment:
I'm not sure exactly what you mean by "masks `const` case"? If you mean it doesn't disambiguate an immutable struct from a const field of a mutable struct, I'm aware of that and this is intentional since we can disambiguate those cases later using reflection.

Reply:
Ok. If you are somehow dealing with it downstream then I don't have complains. Just to check with you; You plan to recalculate idx and test for const qualifier downstream path during reflection. Did I get that right ?